container.go

```diff
@@ -71,7 +72,7 @@ func (c *container) Spec() (*specs.Spec, error) {
 func (c *container) Delete(ctx context.Context) (err error) {
 	// TODO: should the client be the one removing resources attached
 	// to the container at the moment before we have GC?
-	if c.c.RootFS != "" {
+	if c.c.RootFS != "" && strings.HasPrefix(c.c.RootFS, "client-tmp-") {
```

Comment:
I think if you use `WithExistingRootFS` in that way it should still get removed. Use cases for `WithExistingRootFS` is probably better left snapshots created by the caller. I can add a comment on the `With*RootFS` functions which highlight these will get removed on container delete.